### PR TITLE
VBLOCKS-981 VBLOCKS-1093 | Updating eventing API per latest specs

### PR DESCRIPTION
**Pull request by charliesantos**
**Contributing to Twilio**

> All third party contributors acknowledge that any contributions they provide will be made under the same open source license that the open source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

### Description

VBLOCKS-981 VBLOCKS-1093 | Updating eventing API per latest specs

### Before review
* [ ] Updated CHANGELOG.md if necessary
* [x] Added unit tests if necessary
* [ ] Updated affected documentation
* [x] Verified locally with `npm test`
* [x] Manually sanity tested running locally
* [x] Ready for review

### Before merge
* [ ] Got one or more +1s
* [ ] Squashed erroneous commits if necessary
* [ ] Re-tested if necessary

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -9,3 +9,4 @@ extension/token.js
 node_modules
 docs
 lib/twilio/constants.js
+nodemon.json
```
Comment (mhuynh5757) [inline .gitignore:12]: What's this for?

File: lib/twilio/call.ts [hunk 2]
```diff
@@ -215,6 +215,12 @@ class Call extends EventEmitter {
    */
   private _mediaStatus: Call.State = Call.State.Pending;
 
+  /**
+   * A map of messages sent via sendMessage API.
+   * The message will be deleted once an 'ack' or an error is received from the server.
+   */
+  private _messages: Map<string, Call.Message> = new Map();
```
Comment (mhuynh5757) [inline lib/twilio/call.ts:222]: I'm assuming the `string` is the `voiceEventSid` of the message? Should we write that down?

File: lib/twilio/call.ts [hunk 3]
```diff
@@ -1094,6 +1094,21 @@ class Call extends EventEmitter {
     }
   }
 
+  /**
+   * Called when the {@link Call} receives an ack from signaling
+   * @param payload
+   */
+  private _onAck = (payload: Record<string, any>): void => {
```
Comment (mhuynh5757) [inline lib/twilio/call.ts:1101]: `_onMessageAck` for clarity?

File: lib/twilio/call.ts [hunk 4]
```diff
@@ -863,19 +863,15 @@ class Call extends EventEmitter {
...
-   *   ahoy: 'world!,
-   * });
-   * ```
-   *
-   * @param messageType - The type of the message to send to Twilio.
-   * @param content - The content of the message to send to Twilio.
+   * TODO
    */
-  sendMessage(messageType: Call.MessageType, content: any) {
+  sendMessage(message: Call.Message): string {
```
Comment (mhuynh5757) [inline lib/twilio/call.ts:868]: Does returning a `Promise` here that resolves when we receive `ack` make sense?

File: lib/twilio/uuid.ts [hunk 5]
```diff
@@ -28,7 +28,7 @@ function generateUuid(): string {
 
   const generateRandomValues: () => string =
     typeof crypto.randomUUID === 'function'
-      ? crypto.randomUUID
+      ? () => crypto.randomUUID!()
```
Comment (mhuynh5757) [inline lib/twilio/uuid.ts:31]: I see, this is where the illegal invocation was occurring? You could also do ```
? crypto.randomUUID?.bind(crypto)
``` but it doesn't make too much of a difference.

**Review by mhuynh5757 — COMMENTED** [review]
Looks really, really good. Left some feedback.

File: .gitignore (hunk 1 above)
Comment (charliesantos) [reply .gitignore:12]: I'm running nodemon locally to watch and auto build the SDK. I didn't want to commit my personal config.

File: lib/twilio/call.ts (hunk 4 above)
Comment (charliesantos) [reply lib/twilio/call.ts:868]: We initially thought of returning a promise once ack is received. However, it's not the pattern that is used in our SDK. We use pub-sub style for every event that we receive from signaling. In this case, we emit `messageSent` once we receive the `ack`.

**Review by mhuynh5757 — APPROVED** [review]

**Pull request merged**